layouts/shortcodes/glossary_definition.html

```diff
@@ -1,29 +1,33 @@
-
+{{- $scratch := newScratch -}}
 {{- $id := .Get "term_id" -}}
 {{- $length := .Get "length" -}}
-{{- $prepend := .Get "prepend" }}
+{{- $prepend := .Get "prepend" -}}
 {{- $glossaryBundle := site.GetPage "page" "docs/reference/glossary" -}}
 {{- $glossaryItems :=  $glossaryBundle.Resources.ByType "page" -}}
 {{- $term_info := $glossaryItems.GetMatch (printf "%s.md" $id ) -}}
 {{- if not $term_info -}}
 {{- errorf "[%s] %q: %q is not a valid glossary term_id, see ./docs/reference/glossary/* for a full list" site.Language.Lang .Page.Path $id -}}
 {{- end -}}
-{{- with $term_info -}}
-{{- if (strings.Contains "short" $length) -}}
-	{{- with .Summary -}}
-	{{- if $prepend }}{{- replace . "<p>" (printf "<P>%s %s" $prepend .) -}}{{ else }}{{- . -}}{{ end -}}
-	{{- else -}}
-	{{- partial "templates/errorthrower.html" (dict "block" "summary" "purpose" .purpose "describes the key term in greater depth, supplementing the short_description") . -}}
-	{{- end -}}
+{{- $scratch.Set "lengthValid" false -}}
+{{- $scratch.Set "showFullDefinition" false -}}
+{{- if eq "short" $length -}}{{- $scratch.Set "lengthValid" true }}{{- end -}}
+{{- if eq "long" $length  -}}{{- $scratch.Set "lengthValid" true }}{{- $scratch.Set "showFullDefinition" true -}}{{- end -}}
+{{- if eq "all" $length   -}}{{- $scratch.Set "lengthValid" true }}{{- $scratch.Set "showFullDefinition" true -}}{{- end -}}
+{{- if not ($scratch.Get "lengthValid") -}}
+{{- errorf "[%s] %q: invalid glossary definition length %q" site.Language.Lang .Page.Path $length -}}
 {{- end -}}
-{{- if (strings.Contains "all|long" $length) -}}
-{{- with .Content -}}
-{{- if $prepend }}
+{{- with $term_info.Content -}}
+{{- if not ($scratch.Get "showFullDefinition") -}}
 {{- $firstPara := index (findRE "(?s)<p>.*?</p>" . 1) 0 -}}
 {{- $firstPara := $firstPara | strings.TrimSuffix "</p>" | strings.TrimPrefix "<p>" -}}
-{{- $first := slicestr $firstPara 0 1 | lower }}
+{{- $first := slicestr $firstPara 0 1 | lower -}}
+{{- if $prepend }}{{- $prepended := printf "<p>%s %s%s</p>"  $prepend $first (slicestr $firstPara 1) -}}{{- $prepended | safeHTML -}}{{ else }}{{- $firstPara | safeHTML -}}{{ end -}}
```

Comment:
Hi @sftim. A couple of suggestions/questions:
- I think you could remove the "scratch" variable and replace with a single variable, `showFullDefinition`.
- I'd like to see the template code indented.
- Is there a reason why the "short" definition is handled differently than the "long" def?
With the "long" def you check if there is a prepend string before cleaning up the content.
I would expect the two strings to act the same. I did not dig into the way the glossary definition strings
are set up on the page. It would be nice if the strings were handled the same way.

Here is what I used to test the page, `concepts/services-networking/ingress.md`:

```
{{- $id := .Get "term_id" -}}
{{- $length := .Get "length" -}}
{{- $prepend := .Get "prepend" -}}
{{- $glossaryBundle := site.GetPage "page" "docs/reference/glossary" -}}
{{- $glossaryItems :=  $glossaryBundle.Resources.ByType "page" -}}
{{- $term_info := $glossaryItems.GetMatch (printf "%s.md" $id ) -}}
{{- $showFullDefinition := false -}}
{{- if not $term_info -}}
	{{- errorf "[%s] %q: %q is not a valid glossary term_id, see ./docs/reference/glossary/* for a full list" site.Language.Lang .Page.Path $id -}}
{{- end -}}
{{- if or (eq "long" $length) (eq "all" $length) -}}
	{{- $showFullDefinition := true -}}
{{- else if (eq "short" $length) -}}
	{{- $showFullDefinition := false -}}
{{- else -}}
	{{- errorf "[%s] %q: invalid glossary definition length %q" site.Language.Lang .Page.Path $length -}}
{{- end -}}
{{- with $term_info.Content -}}
	{{- if not $showFullDefinition -}}
		{{- $firstPara := index (findRE "(?s)<p>.*?</p>" . 1) 0 -}}
		{{- $firstPara := $firstPara | strings.TrimSuffix "</p>" | strings.TrimPrefix "<p>" -}}
		{{- $first := slicestr $firstPara 0 1 | lower -}}
		{{- if $prepend -}}
			{{- $prepended := printf "<p>%s %s%s</p>"  $prepend $first (slicestr $firstPara 1) -}}
			{{- $prepended | safeHTML -}}
		{{- else -}}
			{{- $firstPara | safeHTML -}}
		{{- end -}}
	{{- else -}}
		{{- if $prepend -}}
			{{- $firstPara := index (findRE "(?s)<p>.*?</p>" . 1) 0 -}}
			{{- $firstPara := $firstPara | strings.TrimSuffix "</p>" | strings.TrimPrefix "<p>" -}}
			{{- $first := slicestr $firstPara 0 1 | lower -}}
			{{- $prepended := printf "<p>%s %s%s</p>"  $prepend $first (slicestr $firstPara 1) -}}
			{{- replace . $firstPara $prepended | safeHTML -}}
		{{- else -}}
			{{- . -}}
		{{- end -}}
	{{- end -}}
{{- end -}}
```

Reply:
With the cluster definition,
What string should I expect to see for the "short" definition vs the "long" or "all" definition:
```
short_description: >
   A set of worker machines, called nodes, that run containerized applications. Every cluster has at least one worker node.
```

Reply:
> I think you could remove the "scratch" variable and replace with a single variable, showFullDefinition.

OK; the if-then-else you suggest does that and it seems pretty clear how it behaves.

> I'd like to see the template code indented.

As I understand it, the whitespace will be copied into the output. Is there a way to indent and not copy whitespace?

> I guess there are limitations on the prepend string, such as, cannot be a sentence?

The prepending behavior is retained from the existing shortcode. I haven't looked at how it works, because I was focused on a defect that's not related to prepending.

Reply:
Re: As I understand it, the whitespace will be copied into the output. Is there a way to indent and not copy whitespace?

Using the double curly brackets with "minus sign and ASCII space character", beginning and end) should remove the trailing whitespace. Does the output look correct?

Reply:
How's the latest revision looking?